### PR TITLE
fix some Jacobian elements for eliminating T integration

### PR DESCRIPTION
**Pull request by zingale**

### CODE REVIEW

**Comment by zingale** [comment]
with these changes, I think that Castro looks good.  I think this is ready to be merged into the original PR.

File: integration/utils/numerical_jacobian.H [hunk 1]
```diff
@@ -104,14 +114,9 @@ void numerical_jac(burn_t& state, JacNetArray2D& jac)
         }
 
         for (int m = 1; m <= neqs; m++) {
-            jac(m, net_itemp) = 0.5_rt * (ydotp(m) - ydotm(m)) / (eps * state.T);
+            jac(m, net_ienuc) = 0.5_rt * (ydotp(m) - ydotm(m)) / (eps * state.e);
```
Comment (maxpkatz) [inline integration/utils/numerical_jacobian.H:117]: This expression (which could be a divide-by-zero if `e` is zero) reminds me that in this new scheme, we really have to be careful that `state.e` is always the actual energy/enthalpy, not a counter starting from zero.

**Comment by zingale** [comment]
we can add an asset checking e == 0

File: integration/VODE/vode_type_simplified_sdc.H [hunk 2]
```diff
@@ -495,11 +495,11 @@ void jac_to_vode(const Real time, burn_t& state,
 
     // d( d(rho X_m)/dt)/dT
     for (int m = 1; m <= NumSpec; m++) {
-        dRdw(SFS+m, iwT) = state.rho * jac_react(m, net_itemp);
+        dRdw(SFS+m, iwT) = state.rho * jac_react(m, net_ienuc);
```
Comment (maxpkatz) [inline integration/VODE/vode_type_simplified_sdc.H:498]: We should change this variable name `iwT` and the comment above

**Comment by maxpkatz** [comment]
We should probably update the definition of `w` in the comments explaining the `dRdw` / `dwdU` scheme

**Review by maxpkatz — APPROVED** [review]

File: integration/VODE/vode_type_simplified_sdc.H (hunk 2 above)
Comment (zingale) [reply integration/VODE/vode_type_simplified_sdc.H]: done

**Pull request merged**